java/arcs/core/data/CapabilityNew.kt

```diff
@@ -0,0 +1,132 @@
+    fun isStricter(other: CapabilityNew) = compare(other) == Comparison.Stricter
+    fun isSameOrStricter(other: CapabilityNew) = compare(other) != Comparison.LessStrict
+
+    abstract fun compare(other: CapabilityNew): Comparison
+
+    /** Capability describing persistence requirement for the store. */
+    data class Persistence(val kind: Kind) : CapabilityNew() {
+        enum class Kind { None, InMemory, OnDisk, Unrestricted }
+
+        override fun compare(other: CapabilityNew): Comparison {
+            require(other is Persistence) { "Cannot compare Persistence with $other" }
+            return when {
+                kind.ordinal < other.kind.ordinal -> Comparison.Stricter
+                kind.ordinal > other.kind.ordinal -> Comparison.LessStrict
+                else -> Comparison.Equivalent
+            }
+        }
+
+        companion object {
+            fun unrestricted() = Persistence(Kind.Unrestricted)
+            fun onDisk() = Persistence(Kind.OnDisk)
+            fun inMemory() = Persistence(Kind.InMemory)
+            fun none() = Persistence(Kind.None)
+        }
+    }
+
+    /** Capability describing retention policy of the store. */
+    sealed class Ttl(count: Int, val isInfinite: Boolean = false) : CapabilityNew() {
+        /** Number of milliseconds for retention, or -1 for infinite. */
+        val millis: Long = count * when (this) {
+            is Millis -> 1
+            is Minutes -> 1 * MILLIS_IN_MIN
+            is Hours -> 60 * MILLIS_IN_MIN
+            is Days -> 60 * 24 * MILLIS_IN_MIN
+            is Infinite -> -1
+        }
+        init {
+            require(count > 0 || isInfinite) {
+                "must be either positive count on infinite, " +
+                    "but got count=$count and isInfinite=$isInfinite"
+            }
+        }
+
+        override fun compare(other: CapabilityNew): Comparison {
+            require(other is Ttl) { "Cannot compare Ttl with $other" }
+            return when {
+                (isInfinite && other.isInfinite) || millis == other.millis -> Comparison.Equivalent
+                isInfinite -> Comparison.LessStrict
+                other.isInfinite -> Comparison.Stricter
+                millis < other.millis -> Comparison.Stricter
+                else -> Comparison.LessStrict
+            }
+        }
+
+        fun calculateExpiration(time: Time): Long {
+            return if (isInfinite) RawEntity.UNINITIALIZED_TIMESTAMP
+            else time.currentTimeMillis + millis
+        }
+
+        data class Millis(val count: Int) : Ttl(count)
+        data class Minutes(val count: Int) : Ttl(count)
+        data class Hours(val count: Int) : Ttl(count)
+        data class Days(val count: Int) : Ttl(count)
+        data class Infinite(val count: Int = TTL_INFINITE) : Ttl(count, true)
+
+        companion object {
+            const val TTL_INFINITE = -1
+            const val MILLIS_IN_MIN: Long = 60 * 1000
```

Comment:
This could be `const val MILLIS_IN_MIN = 60 * 1000L` and then you don't need the type.

Reply:
done, thanks!